lib/realtime/repo.ex

```diff
@@ -38,13 +38,26 @@ defmodule Realtime.Repo do
     |> result_to_single_struct(result_struct)
   end
 
+  @doc """
+  Inserts a given changeset into the database and converts the result into a given struct
+  """
+  @spec insert(DBConnection.conn(), Ecto.Changeset.t(), module()) ::
+          {:ok, struct()} | {:error, any()} | Ecto.Changeset.t()
+  def insert(conn, changeset, result_struct) do
+    with {query, args} <- insert_query_from_changeset(changeset) do
```

Comment:
w`insert_query_from_changeset` at the moment when invalid returns `changeset`, it would be better (and matching this type spec) that it returns `{:error, changeset}`!